From a0d6c917b1a4aa4c30f41990e7fd3b5bb9a324e6 Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Sat, 20 May 2017 14:41:43 -0700
Subject: [PATCH] Respect COMMIT_COUNT when set

This allows setting the commit count even when git is not
installed (or the repository is partial checkout or not even a git
repository at all).
---
 build.proj | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/build.proj b/build.proj
index c4717c4..a7916bc 100644
--- a/build.proj
+++ b/build.proj
@@ -21,8 +21,14 @@
   <Target Name="MakeVersionProps">
     <MakeDir Condition="!Exists('obj')"
              Directories="obj" />
-    <Exec Command="git rev-list --count HEAD" 
-          ConsoleToMSBuild="true">
+
+    <PropertyGroup>
+      <GitInfoCommitCount Condition="'$(COMMIT_COUNT)' != ''">$(COMMIT_COUNT)</GitInfoCommitCount>
+    </PropertyGroup>
+
+    <Exec Command="git rev-list --count HEAD"
+          ConsoleToMSBuild="true"
+          Condition="'$(GitInfoCommitCount)' == ''">
       <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitCount" />
     </Exec>
 
-- 
1.8.3.1

